src/main/java/org/roaringbitmap/buffer/ImmutableRoaringBitmap.java

```diff
@@ -1026,6 +1019,53 @@ public boolean equals(Object o) {
     return false;
   }
 
+  /**
+   * Returns true if the other bitmap has no more than tolerance bits
+   * differing from this bitmap. The other may be transformed into a bitmap equal
+   * to this bitmap in no more than tolerance bit flips if this method returns true.
+   *
+   * @param other the bitmap to compare to
+   * @param tolerance the maximum number of bits that may differ
+   * @return true if the number of differing bits is smaller than tolerance
+   */
+  public boolean isHammingSimilar(ImmutableRoaringBitmap other, int tolerance) {
+    final int size1 = highLowContainer.size();
+    final int size2 = other.highLowContainer.size();
+    int pos1 = 0;
+    int pos2 = 0;
+    int budget = tolerance;
+    while(budget >= 0) {
+      short key1 = this.highLowContainer.getKeyAtIndex(pos1);
```

Comment:
I definitely misunderstood what you were getting at in your original comment, and actually prefer the original implementation to 1057515. Your implementation is sophisticated and I wouldn't have thought to implement the logic that way. I would prefer that either you commit 

`while(budget >= 0 && pos1 < size1 && pos2 < size2) {
      final int key1 = Util.toIntUnsigned(this.highLowContainer.getKeyAtIndex(pos1));
      final int key2 = Util.toIntUnsigned(other.highLowContainer.getKeyAtIndex(pos2));
      final Container c1 = highLowContainer.getContainerAtIndex(pos1);
      final Container c2 = other.highLowContainer.getContainerAtIndex(pos2);
      budget -= ((Integer.signum(key1 - key2) >>> 31) + ((Integer.signum(key1 - key2) + 1) & 0x1)) * c1.getCardinality();
      budget -= ((Integer.signum(key2 - key1) >>> 31) + ((Integer.signum(key1 - key2) + 1) & 0x1)) * c2.getCardinality();
      budget += ((Integer.signum(key1 - key2) + 1) & 0x1) * 2 * c1.andCardinality(c2);
      pos1 += (Integer.signum(key1 - key2) >>> 31) + ((Integer.signum(key1 - key2) + 1) & 0x1);
      pos2 += (Integer.signum(key2 - key1) >>> 31) + ((Integer.signum(key1 - key2) + 1) & 0x1);
    }`

or we revert to the original implementation and back out 1057515. 

The question is whether you'd prefer to be slightly faster to verify that a bitmap is _not_ hamming similar, and avoid calling andCardinality whenever possible, at the expense of being branchy, or if you'd prefer to verify that a bitmap _is_ hamming similar as fast as possible, at the expense of being slower to rule bitmaps out by calling andCardinality more often than necessary.  Being selfish, I need the second use case, and your implementation may well speed up my code that I want to utilise this method from. A benchmark would definitely help decide.